### ISSUE TITLE
About reproduced results

### ISSUE THREAD

**Issue by YAOSL98** [saksham-s/SparseDet#2]
Your paper has been very helpful to me, but I am unable to reproduce the results mentioned in the paper. For example, in the splits5_50% setting, my experimental result for AP50 only reaches 64, whereas in your paper it is 74. In the splits4_extreme setting, my reproduced result is approximately 69, while in your paper it is 75.6. I have not changed any parameter values in your code. Could you help me speculate on the reasons why my experimental results are lower?

**Comment by YAOSL98** [comment]
![image](https://github.com/saksham-s/SparseDet/assets/68221964/6dd02fcc-a10d-4292-9ffe-c40ae0c76767)

Here are the results from my experiments under the splits4_extreme setting.

**Comment by saksham-s** [comment]
Hi, I just reran both the experiments for split4_extreme as well as split5_50p
 with the splits and code uploaded in this repo and got 75.6712 and 73.9923 AP50 respectively which is around what's reported in the paper (there is some non-deterministic behaviour). I am not sure why you are getting a very different number. Are you sure you have not changed anything?

I am also attaching my current environment's requirements file in case their is some issue with your environment.
[requirements.txt](https://github.com/user-attachments/files/16159858/requirements.txt)

**Comment by YAOSL98** [comment]
您好！您发给我的信件已经收到,我将尽快处理

**Comment by YAOSL98** [comment]
I am sure I did not modify any code. I used the configuration commands from the README. The only difference is that I generated the JSON file for inference myself (using the VOC2007 test dataset for inference). This is the JSON file I generated. Could you please help me check if it's correct, or provide the JSON file you used for testing? Thanks a lot
[voc_test.json](https://github.com/user-attachments/files/16186351/voc_test.json)

**Comment by YAOSL98** [comment]
> Hi, I just reran both the experiments for split4_extreme as well as split5_50p with the splits and code uploaded in this repo and got 75.6712 and 73.9923 AP50 respectively which is around what's reported in the paper (there is some non-deterministic behaviour). I am not sure why you are getting a very different number. Are you sure you have not changed anything?
> 
> I am also attaching my current environment's requirements file in case their is some issue with your environment. [requirements.txt](https://github.com/user-attachments/files/16159858/requirements.txt)
[voc_test.json](https://github.com/user-attachments/files/16186452/voc_test.json)

I am sure I did not modify any code. I used the configuration commands from the README. The only difference is that I generated the JSON file for inference myself (using the VOC2007 test dataset for inference). This is the JSON file I generated. Could you please help me check if it's correct, or provide the JSON file you used for testing? Thanks a lot

**Comment by YAOSL98** [comment]
> Hi, I just reran both the experiments for split4_extreme as well as split5_50p with the splits and code uploaded in this repo and got 75.6712 and 73.9923 AP50 respectively which is around what's reported in the paper (there is some non-deterministic behaviour). I am not sure why you are getting a very different number. Are you sure you have not changed anything?
> 
> I am also attaching my current environment's requirements file in case their is some issue with your environment. [requirements.txt](https://github.com/user-attachments/files/16159858/requirements.txt)

I am sure I did not modify any code. I used the configuration commands from the README. The only difference is that I generated the JSON file for inference myself (using the VOC2007 test dataset for inference). This is the JSON file I generated. Could you please help me check if it's correct, or provide the JSON file you used for testing? Thanks a lot
[voc_test.json](https://github.com/user-attachments/files/16186458/voc_test.json)

**Comment by saksham-s** [comment]
I have added my test json to the google drive link where the splits are. Try to use it and see if that works. Given that I register the jsons as coco instances even for VOC, updating the json file to the one I uploaded might just solve the issue.

**Comment by YAOSL98** [comment]
> I have added my test json to the google drive link where the splits are. Try to use it and see if that works. Given that I register the jsons as coco instances even for VOC, updating the json file to the one I uploaded might just solve the issue.

Based on the JSON file you provided, I found the performance issue. The number of GPUs I used is 2 (while it is 4 in your paper). After increasing the value of SOLVER.IMS_PER_BATCH, the performance improved significantly. Again, thank you for your patient help and your excellent paper.

**Comment by epistimi22** [comment]
@saksham-s   The same issue also troubles me. I reproduced your experiment on split4_hard dataset following your settings in README except num_gpu. You used 4 gpus while I used 2, but IMS_PER_BATCH is 8. After 18K iters training, the results on test data is 74.450 AP50 (81.50 in your paper).
I also generated the json file of  test data, and I noticed you've uploaded the voc_test.json file. So I checked these two files and I found that there are 14976 objects in my json file wheares 12032 in yours. But according to the xml files, there should be 14976 objects in test data. 
I'm not sure whether the difference in our results is due to the difference in the number of objects in voc_test json file
[Uploading voc_test.json…]()
.

**Comment by YAOSL98** [comment]
您好！您发给我的信件已经收到,我将尽快处理

**Comment by saksham-s** [comment]
Hi @epistimi22 I am not sure why there is a difference there. Can you try once with running with same gpus and annotation file. Also I used https://github.com/yukkyo/voc2coco as a reference to convert the voc files to coco format.

**Comment by epistimi22** [comment]
> Hi @epistimi22 I am not sure why there is a difference there. Can you try once with running with same gpus and annotation file. Also I used https://github.com/yukkyo/voc2coco as a reference to convert the voc files to coco format.

I ran your code with _your test file_ and the result was **81.67** AP50. It's quite weird and it seems to suggest that the test file is the key reason for the differences in our results. I will check the generation process in https://github.com/yukkyo/voc2coco. Thanks for your help!

**Comment by epistimi22** [comment]
@saksham-s Both test files of mine and @YAOSL98  are 14976 objects and I ran your voc2coco.py to generate test json file like you did, there were 14976 objects too. 
In fact, if you calculate the total number of bounding boxes in test **xml files** based on _VOCdevkit/VOC2007/ImageSets/Main/test.txt_, you will find the total obejects/bounding boxes are **14976**. So could you please re-generate your test json file with an original VOC07+12 dataset?

**Comment by saksham-s** [comment]
I tried generating with re-downloading the VOCdevkit and got the same json as what I downloaded. Finally I tried with recloning the voc2coco and repo and was able to get the same number as you got. On further investigation I was able to identify why this discrepancy exists. If you see Fig 3 and corresponding text in the original [VOC Paper ](http://host.robots.ox.ac.uk/pascal/VOC/pubs/everingham10.pdf) they talk about excluding images with "difficult" tag from evals. Given I am using coco registration for this dataset I had to remove those myself while converting and had added a line to do so in my voc2coco.py. Irrespective if you want to use the newer json with difficult as long as you evaluate all methods with it, that should be reasonable too.
